pkg/operators/localmanager/localmanager.go

```diff
@@ -417,6 +425,160 @@ func (l *localManagerTrace) EnrichEvent(ev any) error {
+	}
+
+	activate := true // TODO: change to false once snapshot_socket is not skipped anymore
+
+	// Check, whether the gadget requested a map from us
+	if t, ok := ctx.GetVar(gadgets.MntNsFilterMapName); ok {
+		if _, ok := t.(*ebpf.Map); ok {
+			ctx.Logger().Debugf("gadget requested map %s", gadgets.MntNsFilterMapName)
+			activate = true
+		}
+	}
+
+	// Check for data sources containing refererences to mntns/netns that we could enrich data for
+	for _, ds := range ctx.GetDataSources() {
+		mntnsFields := ds.GetFieldsWithTag("type:gadget_mntns_id")               // TODO: const
+		netnsFields := ds.GetFieldsWithTag("type:gadget_netns_id", "name:netns") // TODO: const
+		if len(mntnsFields) == 0 && len(netnsFields) == 0 {
+			continue
+		}
+
+		ctx.Logger().Debugf("ds found: %q", ds.Name())
+
+		activate = true
+
+		var err error
+
+		var mntnsField datasource.FieldAccessor
+		var netnsField datasource.FieldAccessor
+
+		for _, f := range mntnsFields {
+			ctx.Logger().Debugf("using mntns enrichment")
+			mntnsField = f
+			mntnsField.SetHidden(true)
+			// We only support one of those per DataSource for now
+			break
+		}
+		for _, f := range netnsFields {
+			ctx.Logger().Debugf("using netns enrichment")
+			netnsField = f
+			netnsField.SetHidden(true)
+			// We only support one of those per DataSource for now
+			break
+		}
+
+		traceInstance.eventWrappers[ds], err = registerAccessors(ds, mntnsField, netnsField)
+		if err != nil {
+			return nil, fmt.Errorf("registering accessors: %w", err)
+		}
+	}
+
+	if !activate {
+		return nil, nil
+	}
+
+	return traceInstance, nil
+}
+
+func (l *localManagerTrace) ParamDescs() params.ParamDescs {
+	return params.ParamDescs{
+		{
+			Key:         ContainerName,
+			Alias:       "c",
+			Description: "Show only data from containers with that name",
+			ValueHint:   gadgets.LocalContainer,
+		},
+		{
+			Key:          Host,
+			Description:  "Show data from both the host and containers",
+			DefaultValue: "false",
+			TypeHint:     params.TypeBool,
+		},
+	}
+}
+
+func (l *localManagerTrace) Prepare(gadgetCtx operators.GadgetContext) error {
+	id := uuid.New()
+	host := false // TODO
+
+	containerSelector := containercollection.ContainerSelector{
+		Runtime: containercollection.RuntimeSelector{
+			ContainerName: "", // TODO l.params.Get(ContainerName).AsString(),
+		},
+	}
+
+	if !host {
+		if l.manager.igManager == nil {
+			return fmt.Errorf("container-collection isn't available")
+		}
+
+		// Create mount namespace map to filter by containers
+		mountnsmap, err := l.manager.igManager.CreateMountNsMap(id.String(), containerSelector)
+		if err != nil {
+			return commonutils.WrapInErrManagerCreateMountNsMap(err)
+		}
+
+		log.Debugf("set mountnsmap for gadget")
+		gadgetCtx.SetVar(gadgets.MntNsFilterMapName, mountnsmap)
+
+		l.mountnsmap = mountnsmap
+	} else if l.manager.igManager == nil {
+		log.Warn("container-collection isn't available: container enrichment and filtering won't work")
+	}
```

Comment:
Then you would introduce a hard dependency on that from the eBPF operator - and we have two implementations (KubeManager + LocalManager). Also, we want to add the possibility to completely deactivate certain operators. I think having this contained in an operator helps keeping the code clean.

> The mount ns map is something that can only be used from eBPF programs
It's probably going to be only eBPF accessing it in this specific case - but for other features, I think other operators might be interested in eBPF maps as well. Weird example: imagine an operator that exposes itself using a JSON/REST interface over HTTP that accepts data on a url like `/api/:gadgetID/:mapName/set?key=abc&val=def` where it would edit map contents that way.

Reply:
> Then you would introduce a hard dependency on that from the eBPF operator 

What is the issue with this? The eBPF operator takes care of loading and attaching eBPF programs to the kernel, why can't it also take care of creating the filtering map that is used by those programs to discard events from the kernel? 

I see that the {Local,Kube}Manager operators are (1) creating the ebpf map for filtering purposes (2) enriching data, do these two operations belong to the same operator? I know both features need the container collection, but other than that I'd say they're independent. 

> and we have two implementations (KubeManager + LocalManager). 

Are those two operator really different? We're trying to close the gap between ig and ig-k8s (see #1403 &  #2100), so probably those two operators are going to finish being a single one. 

> Also, we want to add the possibility to completely deactivate certain operators. I think having this contained in an operator helps keeping the code clean.

It's a good point. Two things:
- Perhaps instead of disabling the whole operator it could be an option inside the ebpf operator? 
- I see a point on disabling some operators, but {Local,Kube}Manager are so important that I don't see a point in disabling them. Running the gadgets without providing any enrichment nor filtering capabilities?

> It's probably going to be only eBPF accessing it in this specific case - but for other features, I think other operators might be interested in eBPF maps as well. Weird example: imagine an operator that exposes itself using a JSON/REST interface over HTTP that accepts data on a url like /api/:gadgetID/:mapName/set?key=abc&val=def where it would edit map contents that way.

How would this work? The ebpf operator puts all maps into the gadget context, and then this "ebpf-map-updater" uses this information to update them? Besides map, which other things should the ebpf operator expose and how will other operators understand this information is available on the context? In other words, what's the contract between different operators?

Reply:
> > Then you would introduce a hard dependency on that from the eBPF operator
> 
> What is the issue with this? The eBPF operator takes care of loading and attaching eBPF programs to the kernel, why can't it also take care of creating the filtering map that is used by those programs to discard events from the kernel?
> 
> I see that the {Local,Kube}Manager operators are (1) creating the ebpf map for filtering purposes (2) enriching data, do these two operations belong to the same operator? I know both features need the container collection, but other than that I'd say they're independent.

Hmm, I think I see it this way: the eBPF operator (or other layer operators) provides the framework / tooling, while data operators provide the additional content / features. IMHO it's best to split that as much as possible. What advantages do you see other than removing this particular abstraction?

> > and we have two implementations (KubeManager + LocalManager).
> 
> Are those two operator really different? We're trying to close the gap between ig and ig-k8s (see #1403 & #2100), so probably those two operators are going to finish being a single one.

Yeah, it would be great if we could avoid duplicate code, but IMHO the combined operator should still be handled independently of the eBPF operator.

> 
> > Also, we want to add the possibility to completely deactivate certain operators. I think having this contained in an operator helps keeping the code clean.
> 
> It's a good point. Two things:
> 
> * Perhaps instead of disabling the whole operator it could be an option inside the ebpf operator?
> * I see a point on disabling some operators, but {Local,Kube}Manager are so important that I don't see a point in disabling them. Running the gadgets without providing any enrichment nor filtering capabilities?

If we can reduce direct dependencies, I think we should. This is especially the case for the Go API that we want to provide - if using IG as a lib, it would be better to import the least amount of code to achieve what you want. Importing for example the KubeManager brings _a lot_ of dependencies on its own (basically the whole k8s ecosystem with tons of other packages) - if you only want to use IG as a tool to load eBPF programs and leverage the tooling / exporters etc, that might be overkill. I know the enrichments are probably our KSP right now, but that doesn't mean that the rest of IG isn't worth being used on its own 😉 
I think the price of having things independent of each other after this refactoring is really low and that's why I'm all for going this way. 😄 But I'm happy to discuss further.

> 
> > It's probably going to be only eBPF accessing it in this specific case - but for other features, I think other operators might be interested in eBPF maps as well. Weird example: imagine an operator that exposes itself using a JSON/REST interface over HTTP that accepts data on a url like /api/:gadgetID/:mapName/set?key=abc&val=def where it would edit map contents that way.
> 
> How would this work? The ebpf operator puts all maps into the gadget context, and then this "ebpf-map-updater" uses this information to update them? Besides map, which other things should the ebpf operator expose and how will other operators understand this information is available on the context? In other words, what's the contract between different operators?

OTTOMH, this could be achieved by exposing a map as a datasource through the eBPF Operator, tagging it using metadata with something like "exported" or "public" and then having the JSON/REST operator that would expose that datasource - this would mean we'd have to extend the DataSource to be able to fetch specific data, which we probably need sooner or later anyway.